src/main/java/picard/fingerprint/HaplotypeProbabilitiesFromContaminatorSequence.java

```diff
@@ -90,16 +104,21 @@ public void addToProbs(final Snp snp, final byte base, final byte qual) {
 
     //a function needed to update the logLikelihoods from the likelihoodMap.
     private void updateLikelihoods() {
-        final double[] ll = new double[Genotype.values().length];
+        final double[] ll = new double[nGeno];
         for (final Genotype contGeno : Genotype.values()) {
             // p(a | g_c) = \sum_g_m { P(g_m) \prod_i P(a_i| g_m, g_c)}
-            ll[contGeno.v] = log10(MathUtil.sum(MathUtil.multiply(this.getPriorProbablities(), likelihoodMap[contGeno.v])));
+            ll[contGeno.v] = Math.log10(MathUtil.sum(MathUtil.multiply(this.getPriorProbablities(), likelihoodMap[contGeno.v])));
         }
         setLogLikelihoods(ll);
     }
 
     @Override
-    public void merge(final HaplotypeProbabilities other) {
+    public HaplotypeProbabilities deepCopy() {
+        return new HaplotypeProbabilitiesFromContaminatorSequence(this);
+    }
+
+    @Override
+    public HaplotypeProbabilities merge(final HaplotypeProbabilities other) {
```

Comment:
This can also use covariant return types

Reply:
:+1: